### PR TITLE
obj-magic: init at unstable-2020-09-20

### PR DESCRIPTION
**Pull request by lorenz**
## Description of changes

Adds obj-magic, a command line tool for manipulating Wavefront OBJ 3d meshes.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ob/obj-magic/package.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+}:
+
+stdenv.mkDerivation {
+  pname = "obj-magic";
+  version = "unstable-2020-09-20";
```
Comment (yunfachi) [inline pkgs/by-name/ob/obj-magic/package.nix:8]:
```suggestion
  version = "0.5-unstable-2020-09-20";
```
I found a version in the [source code](https://github.com/tapio/obj-magic/blob/f25c9b78cee6529a3295ed314d1c200677dc56c0/src/obj-magic.cpp#L19).

https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md

File: pkgs/by-name/ob/obj-magic/package.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  '';
+
+  installPhase = ''
+    runHook preInstall
+    install -D obj-magic $out/bin/obj-magic
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "obj-magic is a command line tool for manipulating Wavefront OBJ 3d meshes";
```
Comment (yunfachi) [inline pkgs/by-name/ob/obj-magic/package.nix:30]: You should remove the package name from the beginning and change `3d` to `3D`

https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#meta-attributes
 * `meta.description` must:
   * Be capitalized.
   * Not start with the package name.
     * More generally, it should not refer to the package name.

**Review by yunfachi — CHANGES_REQUESTED** [review]

File: pkgs/by-name/ob/obj-magic/package.nix [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+    install -D obj-magic $out/bin/obj-magic
+    runHook postInstall
+  '';
+
+  meta = with lib; {
```
Comment (yunfachi) [inline pkgs/by-name/ob/obj-magic/package.nix:29]:
```suggestion
  meta = {
```

File: pkgs/by-name/ob/obj-magic/package.nix [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    description = "A command line tool for manipulating Wavefront OBJ 3D meshes";
+    homepage = "https://github.com/tapio/obj-magic";
+    license = licenses.gpl3Only;
+    maintainers = with maintainers; [ lorenz ];
```
Comment (yunfachi) [inline pkgs/by-name/ob/obj-magic/package.nix:33]:
```suggestion
    license = lib.licenses.gpl3Only;
    maintainers = with lib.maintainers; [ lorenz ];
```

File: pkgs/by-name/ob/obj-magic/package.nix (hunk 3 above)
Comment (yunfachi) [reply pkgs/by-name/ob/obj-magic/package.nix:29]: https://github.com/NixOS/nixpkgs/issues/292468

File: pkgs/by-name/ob/obj-magic/package.nix [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  '';
+
+  installPhase = ''
+    runHook preInstall
+    install -D obj-magic $out/bin/obj-magic
+    runHook postInstall
+  '';
+
+  meta = {
+    description = "A command line tool for manipulating Wavefront OBJ 3D meshes";
```
Comment (SuperSandro2000) [inline pkgs/by-name/ob/obj-magic/package.nix:30]:
```suggestion
    description = "Command line tool for manipulating Wavefront OBJ 3D meshes";
```

**Pull request merged**